client/go/outline/electron/main.go

```diff
@@ -115,11 +122,11 @@ func main() {
 	if len(*args.transportConfig) == 0 {
 		printErrorAndExit(platerrors.PlatformError{Code: platerrors.InvalidConfig, Message: "transport config missing"}, exitCodeFailure)
 	}
-	clientResult := outline.NewClient(*args.transportConfig)
-	if clientResult.Error != nil {
-		printErrorAndExit(clientResult.Error, exitCodeFailure)
+
+	client, err := newOutlineClient(*args.transportConfig, *args.adapterIP, *args.adapterIndex)
```

Comment:
This code says that the adapter is required. It's not.

I'd like to see the code read in a way that checks the presence of the adapter, calling different functions (the old call or the new call with the adapter).

The new call with the adapter should return unsupported error for non-implemented platforms.

That will more clearly communicate the higher level logic of this code.

Reply:
Updated